src/dotnet-svcutil/lib/src/FrameworkFork/Microsoft.Xml/Xml/AsyncHelper.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Xml {
+				using System;
+				
+    internal static class AsyncHelper {
+        public static readonly Task DoneTask = Task.FromResult(true);
+
+        public static readonly Task<bool> DoneTaskTrue = Task.FromResult(true);
+
+        public static readonly Task<bool> DoneTaskFalse = Task.FromResult(false);
+
+        public static readonly Task<int> DoneTaskZero = Task.FromResult(0);
+                        
+        public static bool IsSuccess(this Task task) {
+            return task.IsCompleted && task.Exception == null;
+        }
+
+        public static Task CallVoidFuncWhenFinish(this Task task, Action func) {
+            if (task.IsSuccess()) {
+                func();
+                return DoneTask;
+            }
+            else {
+                return _CallVoidFuncWhenFinish(task, func);
+            }
+        }
+
+        private static async Task _CallVoidFuncWhenFinish(this Task task, Action func) {
+            await task.ConfigureAwait(false);
+            func();
+        }
+
+        public static Task<bool> ReturnTaskBoolWhenFinish(this Task task, bool ret) {
+            if (task.IsSuccess()) {
```

Comment:
The compiler caches bool tasks when a method is async. Convert to async and directly await the passed in Task. There's actually a bug here. If the task throws, if it's already completed this method returns false, if it isn't yet completed, it will throw. Not sure if that's an issue or not.